### PR TITLE
[ci] Set artifact expire date for all jobs.

### LINKED ISSUE #8735: [ci] [gitlab] `expire_in` not set in all targets.
As of today, the CI build produces some "keep forever" artifacts. IMHO, there is not point in doing so, and we should add the corresponding `expire_in` to all targets.

### PR DESCRIPTION
**Pull request by ejgallego**
Closes #8735 .

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -18,6 +18,9 @@ variables:
   OPAM_VARIANT: ""
   GIT_DEPTH: "10"
 
+artifacts:
+  expire_in: 3 days
```
Comment (SkySkimmer) [inline .gitlab-ci.yml]: Why not 1 week and remove the redundant settings?

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
Logs should be forever just like the main build log.

File: .gitlab-ci.yml (hunk 1 above)
Comment (ejgallego) [reply .gitlab-ci.yml]: I prefer the information to be more "local".

**Comment by ejgallego** [comment]
> Logs should be forever just like the main build log.

Ok, not a fan of this but will fix.

File: .gitlab-ci.yml (hunk 1 above)
Comment (ejgallego) [reply .gitlab-ci.yml]: Will fix too.

Comment (ejgallego) [reply .gitlab-ci.yml]: Oh actually the global setting doesn't work:
```
jobs:artifacts config contains unknown keys: expire_in
You can also test your .gitlab-ci.yml in CI Lint
```

**Comment by ejgallego** [comment]
Rebased; test-suite logs kept forever. It seems that it is not possible to set a global expire setting [except in the instance config]

**Comment by ejgallego** [comment]
Turns out `expire_in: never` is not supported either; used 2 years.

**Comment by SkySkimmer** [comment]
>Turns out expire_in: never is not supported either

Just don't put an `expire_in` key.

**Comment by ejgallego** [comment]
> Just don't put an `expire_in` key.

The point of #8735 was indeed to have all targets documented; IMO that is better than relying on the unknown instance-wide setting.

This could be moved to a different gitlab [unlikely I know] and then logs could be lost.

Thus I prefer to keep it like this [we can put 5 years if you want?]

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -385,6 +388,7 @@ doc:refman:dune:
   artifacts:
     paths:
       - _build/default/doc/sphinx_build/html
+    expire_in: 2 months
```
Comment (SkySkimmer) [inline .gitlab-ci.yml]: Why not the same as dune-ci-template?

**Comment by SkySkimmer** [comment]
>Thus I prefer to keep it like this [we can put 5 years if you want?]

IMO I see no gain, but 2 years should be OK so we can merge if you really want it.

**Comment by ejgallego** [comment]
> IMO I see no gain, but 2 years should be OK so we can merge if you really want it.

Well I consider that ⊥⊆ date , so I prefer the more defined entry, but also I don't feel strongly about it; I dunno.

File: .gitlab-ci.yml (hunk 2 above)
Comment (ejgallego) [reply .gitlab-ci.yml]: I used 2 months for docs and 1 month for binaries. I dunno, these values are arbitrary.

**Comment by maximedenes** [comment]
@SkySkimmer can you merge if ready?

File: .gitlab-ci.yml (hunk 2 above)
Comment (SkySkimmer) [reply .gitlab-ci.yml]: >I used 2 months for docs and 1 month for binaries. I dunno, these values are arbitrary.

Please pick one and use it for both.

**Comment by ejgallego** [comment]
No rush on this.

**Comment by ejgallego** [comment]
What we do with this ? Forever is not supported so we can either keep 5 years or I can just leave a comment and not `expire_in` attribute.

**Review by SkySkimmer — CHANGES_REQUESTED** [review]
No need for a different delay for docs vs binaries IMO

**Comment by SkySkimmer** [comment]
>What we do with this ? Forever is not supported so we can either keep 5 years or I can just leave a comment and not expire_in attribute.

I'll take whichever is implemented when you fix my change request.

**Comment by ejgallego** [comment]
> I'll take whichever is implemented when you fix my change request.

Ok thanks, updated. I hope I got it right.

File: .gitlab-ci.yml (hunk 2 above)
Comment (SkySkimmer) [reply .gitlab-ci.yml]: No need to repeat it when it's the same as the template.

Comment (ejgallego) [reply .gitlab-ci.yml]: well seen, let's see now.

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**